Lib/test/test_tkinter/test_text.py

```diff
@@ -27,20 +27,38 @@ def test_debug(self):
     def test_search(self):
         text = self.text
 
-        # pattern and index are obligatory arguments.
         self.assertRaises(tkinter.TclError, text.search, None, '1.0')
         self.assertRaises(tkinter.TclError, text.search, 'a', None)
         self.assertRaises(tkinter.TclError, text.search, None, None)
-
-        # Invalid text index.
         self.assertRaises(tkinter.TclError, text.search, '', 0)
 
-        # Check if we are getting the indices as strings -- you are likely
-        # to get Tcl_Obj under Tk 8.5 if Tkinter doesn't convert it.
         text.insert('1.0', 'hi-test')
         self.assertEqual(text.search('-test', '1.0', 'end'), '1.2')
         self.assertEqual(text.search('test', '1.0', 'end'), '1.3')
 
+        text.delete('1.0', 'end')
+        text.insert('1.0',
+            'This is a test. This is only a test.\n'
+            'Another line.\n'
+            'Yet another line.')
+
+        result = text.search('line', '1.0', 'end', nolinestop=True, regexp=True)
+        self.assertEqual(result, '2.8')
+
+        all_res = text.search_all('test', '1.0', 'end')
+        self.assertIsInstance(all_res, tuple)
+        self.assertGreaterEqual(len(all_res), 2)
+        self.assertEqual(str(all_res[0]), '1.10')
+        self.assertEqual(str(all_res[1]), '2.8')
+
+
+        overlap_res = text.search_all('test', '1.0', 'end', overlap=True)
```

Comment:
For testing `overlap`, please use a pattern that produces overlapping substrings. Test with and without `overlap=True` to show that this affects the result.

I think that it is better to extract the `search_all()` tests into a separate method -- anyway they search in a different text.